API/c/Src/analog_in.c

```diff
@@ -0,0 +1,65 @@
+#include "analog_in.h"
+
+// ADC Reading constants
+const int REFRESH_PERIOD = 10;
+const int ADC_MAX = 4096;
+const float ADC_RATIO = 3.3 / 4096;
+
+void analogin_init(adc_pin_name pin, adc_handle* handle) {
```

Comment:
I'm seeing something about `HAL_ADC_MspInit()` and implementing that. It make seense due to all the `__*` functions that shouldn't be exposed at this layer, but I haven't looked into it.

Related section in 6.2.2

```
1. Initialize the ADC low level resources by implementing the HAL_ADC_MspInit():
a. Enable the ADC interface clock using __HAL_RCC_ADC_CLK_ENABLE()
b. ADC pins configuration
Enable the clock for the ADC GPIOs using the following function:
__HAL_RCC_GPIOx_CLK_ENABLE()
Configure these ADC pins in analog mode using HAL_GPIO_Init()
c. In case of using interrupts (e.g. HAL_ADC_Start_IT()) 
HAL ADC Generic Driver UM1905
92/1967 DocID027932 Rev 3
Configure the ADC interrupt priority using HAL_NVIC_SetPriority()
Enable the ADC IRQ handler using HAL_NVIC_EnableIRQ()
In ADC IRQ handler, call HAL_ADC_IRQHandler()
d. In case of using DMA to control data transfer (e.g. HAL_ADC_Start_DMA())
Enable the DMAx interface clock using
__HAL_RCC_DMAx_CLK_ENABLE()
Configure and enable two DMA streams stream for managing data transfer
from peripheral to memory (output stream)
Associate the initialized DMA handle to the CRYP DMA handle using
__HAL_LINKDMA()
Configure the priority and enable the NVIC for the transfer complete interrupt
on the two DMA Streams. The output stream should have higher priority
than the input stream. 
```

Reply:
Can confirm. All LL calls should be in a user implemented ADC_MspInit. Line 293 here calls the MspIniit in the ADC_Init and line 381 shows that it should be user implemented. If we do a MspIniit, we need a MspDeIniti as well.

http://www.disca.upv.es/aperles/arm_cortex_m3/llibre/st/STM32F439xx_User_Manual/stm32f4xx__hal__adc_8c_source.html

This is F4, but it should be fine for F7

Reply:
Implemented MSP deinit